From 19688cbe2ca10921a499f3fa265928dca54cf98d Mon Sep 17 00:00:00 2001
From: Alejandro Joya <alejandro.joya.cruz@intel.com>
Date: Wed, 4 Nov 2015 19:06:23 -0600
Subject: [PATCH 2/2] app-install: implement multiple set of smack-rules

If it's need it could create load multiple set of smack rules
related with the privileges.
It wouldn't affect the case that only the default set of rules is need it.

Signed-off-by: Alejandro Joya <alejandro.joya.cruz@intel.com>
---
 src/common/service_impl.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/common/service_impl.cpp b/src/common/service_impl.cpp
index 7fd621c..ae305d3 100644
--- a/src/common/service_impl.cpp
+++ b/src/common/service_impl.cpp
@@ -338,6 +338,12 @@ int appInstall(const app_inst_req &req, uid_t uid)
         LogDebug("Adding Smack rules for new appId: " << req.appId << " with pkgId: "
                 << req.pkgId << ". Applications in package: " << pkgContents.size());
         SmackRules::installApplicationRules(req.appId, req.pkgId, pkgContents);
+	/*Setup for privileges custom rules*/
+	 LogDebug("Adding Smack rules for new appId: " << req.appId << " with pkgId: "
+                << req.pkgId << ". Applications in package: " << pkgContents.size()
+		<< " and Privileges");
+	SmackRules::installApplicationPrivilegesRules(req.appId, req.pkgId,
+	    pkgContents,req.privileges);
     } catch (const SmackException::Base &e) {
         LogError("Error while applying Smack policy for application: " << e.DumpToString());
         return SECURITY_MANAGER_API_ERROR_SETTING_FILE_LABEL_FAILED;
-- 
2.1.0

